include/aws/iotdevice/iotdevice.h

```diff
@@ -17,6 +17,9 @@ enum aws_iotdevice_error {
     AWS_ERROR_IOTDEVICE_DEFENDER_UNSUPPORTED_REPORT_FORMAT,
     AWS_ERROR_IOTDEVICE_DEFENDER_REPORT_SERIALIZATION_FAILURE,
 
+    AWS_ERROR_IOTDEVICE_SECUTRE_TUNNELING_INVALID_STREAM,
+    AWS_ERROR_IOTDEVICE_SECUTRE_TUNNELING_INCORRECT_MODE,
```

Comment:
Consideration: Incorrect proxy mode based on a failed handshake's output maybe shouldn't be a CRT error. Probably just want to generically represent a "handshake failure," and let the internal logging show why. Definitely distinguish between the WebSocket 4xx responses (client messed something up likely) and the 5xx failure responses (client should or can retry).